server/routes.js

```diff
@@ -131,13 +131,15 @@ function getDomainSlug(publisherConfig, hostName) {
 function withConfigPartial(
   getClient,
   logError,
-  publisherConfig = require("./publisher-config")
+  publisherConfig = require("./publisher-config"),
+  configWrapper = config => config
 ) {
   return function withConfig(f, staticParams) {
     return function (req, res, next) {
       const client = getClient(req.hostname);
       return client
         .getConfig()
+        .then(config => configWrapper(config))
```

Comment:
Shouldnt this be a await or a promise?

Reply:
Yes, it returns a Promise which is handled by the next `then`

Reply:
I meant shouldnt it be like this. `await config wrapper(config)`

Reply:
I don't think so. Nested promises return the final resolved value.

Run the following code:

 // Checking nested Promise behaviour
 const nestedPromise = () => {
   return new Promise((resolve, reject) => {
     return resolve(new Promise((resolve, reject) => resolve(5)));
   });
 }

 const value = nestedPromise().then(result => {
   console.log("Result =>", result);
   return result;
  });

Reply:
if the function passed to the then function returns a `Promise`, as it happens here when the `configWrapper` function is called, the next `then` in the promise chain waits till it is resolved, in this case `await` inside the resolve function is not necessary

ref: https://javascript.info/promise-chaining#summary

Reply:
But this is true only if the function returns a promise. Say we dont return a promise?

Reply:
We don't need to do any of it. Usually this is how we implement it. Please let me know how u want me to update the current implementation. I will make the changes